src/Tools/GenerateDocumentationAndConfigFiles/Program.cs

```diff
@@ -352,7 +392,7 @@ void createAnalyzerSarifFile()
 
                 var directory = Directory.CreateDirectory(analyzerSarifFileDir);
                 var fileWithPath = Path.Combine(directory.FullName, analyzerSarifFileName);
-
+                // TODO: Validate only.
```

Comment:
Stale comment?

Reply:
Nope. This is still a TODO.
This specific one is using a JsonWriter and a StreamWriter and writing to the file directly. I haven't thought yet on how to fix this with minimal code. One approach would be updating JsonWriter to write to a stringbuilder instead, or alternatively having some sort of a common interface between an stringbuilder writer and a streamwriter.

What do you think would be a good approach here?

Reply:
One possible approach could be that when `validateOnly` is true, write to a file in a temp/intermediate directory and then compare the contents of two files.